File: frontend/src/types/LabWork.ts
import type { Equipment } from './Equipment';

export enum LabWorkStatus {
    DRAFT = 'DRAFT',
    PUBLISHED = 'PUBLISHED',
    ARCHIVED = 'ARCHIVED'
}

export interface LabWork {
    id: number;
    title: string;
    description?: string;
    authorUsername: string;
    requiredEquipment: Equipment[];
    status: LabWorkStatus;
    createdAt: string;
    updatedAt: string;
}

export interface LabWorkCreateRequest {
    title: string;
    description?: string;
    equipmentIds: number[];
}

export interface LabWorkUpdateRequest {
    title?: string;
    description?: string;
    equipmentIds?: number[];
    status?: LabWorkStatus;
}

File: backend/src/main/java/com/example/labsystem/domain/labwork/LabWork.java
package com.example.labsystem.domain.labwork;

import com.example.labsystem.domain.lab.Equipment;
import com.example.labsystem.domain.user.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * Entity для лабораторних робіт (аналог Projects з microlab_v2).
 * Лабораторна робота містить опис, необхідне обладнання та автора.
 */
@Entity
@Table(name = "lab_works")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LabWork {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    private User author;

    @ManyToMany
    @JoinTable(
            name = "labwork_equipment",
            joinColumns = @JoinColumn(name = "labwork_id"),
            inverseJoinColumns = @JoinColumn(name = "equipment_id")
    )
    @Builder.Default
    private List<Equipment> requiredEquipment = new ArrayList<>();

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private LabWorkStatus status = LabWorkStatus.DRAFT;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
